src/web/error.rs

```diff
@@ -134,4 +160,66 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    fn check_404_page_content_not_semver_version() {
+        wrapper(|env| {
+            env.fake_release().name("dummy").create()?;
+            let page =
+                kuchiki::parse_html().one(env.frontend().get("/dummy/not-semver").send()?.text()?);
+            assert_eq!(page.select("#crate-title").unwrap().count(), 1);
+            assert_eq!(
+                page.select("#crate-title")
+                    .unwrap()
+                    .next()
+                    .unwrap()
+                    .text_contents(),
+                "The requested version does not exist",
+            );
+
+            Ok(())
+        });
+    }
+
+    #[test]
+    fn check_404_page_content_nonexistent_version() {
+        wrapper(|env| {
+            env.fake_release().name("dummy").version("1.0.0").create()?;
+            let page = kuchiki::parse_html().one(env.frontend().get("/dummy/2.0").send()?.text()?);
+            assert_eq!(page.select("#crate-title").unwrap().count(), 1);
+            assert_eq!(
+                page.select("#crate-title")
+                    .unwrap()
+                    .next()
+                    .unwrap()
+                    .text_contents(),
+                "The requested version does not exist",
+            );
+
+            Ok(())
+        });
+    }
+
+    #[test]
+    fn check_404_page_content_any_version_all_yanked() {
+        wrapper(|env| {
+            env.fake_release()
+                .name("dummy")
+                .version("1.0.0")
+                .yanked(true)
+                .create()?;
+            let page = kuchiki::parse_html().one(env.frontend().get("/dummy/*").send()?.text()?);
+            assert_eq!(page.select("#crate-title").unwrap().count(), 1);
+            assert_eq!(
+                page.select("#crate-title")
+                    .unwrap()
+                    .next()
+                    .unwrap()
+                    .text_contents(),
+                "The requested version does not exist",
```

Comment:
Eventually it would be nice to instead say 'all versions of this crate are yanked', which makes more sense since this didn't request a specific version. But that can be a follow-up PR.